tensorboard/program.py

```diff
@@ -218,6 +225,40 @@ def main(self, ignored_argv=('',)):
       sys.stderr.flush()
       return -1
 
+  def _register_info(self, server):
+    server_url = urllib.parse.urlparse(server.get_url())
+    info = manager.TensorboardInfo(
+        version=version.VERSION,
+        start_time=int(time.time()),
+        port=server_url.port,
+        pid=os.getpid(),
+        path_prefix=self.flags.path_prefix,
+        logdir=self.flags.logdir,
+        db=self.flags.db,
+        cache_key=manager.cache_key(
+            working_directory=os.getcwd(),
+            arguments=sys.argv[1:],
+        ),
+    )
+
+    @atexit.register
+    def clean_up():
+      manager.remove_info_file()
+
+    def handle_sigterm(signal_number, stack_frame):
```

Comment:
This doesn't exclusively relate to the rest of this function, so perhaps better to separate concerns/shorten the function and register this separately with a comment that this helps ensure atexit functions get a chance to run for SIGTERM.  E.g. the `--db_import` mode registers an atexit function, so arguably there's a reason to have the SIGTERM handler independent of the new atexit that you're adding.